### ISSUE TITLE
Suite sometimes doesn't refresh or show all transactions

### ISSUE THREAD

**Issue by uj** [trezor/trezor-suite#3848]
Suite sometimes doesn't refresh or show all transactions.  I have read about this from several other people on forums, similar issues, and today it happened to me.  I sent btc from coinbase to trezor btc address.  A couple minutes later, I sent another identical amount of btc to the exact same trezor btc address, so that I would have 2 transactions.

Both transactions showed up in Suite as pending, just as expected.  After some time, once they were confirmed, 1 of the transactions showed up in Suite, but the other transaction disappeared completely.  However, the graph up above the transaction list showed the proper amount (the sum of both identical transactions), even though only 1 transaction showed in the list.

I tried clicking back and forth to other coins, back and forth from dashboard to accounts.  That did not fix it.
I closed and re-opened suite a few times, still it did not fix it.

If I used the search bar right above the transaction list, and put a dot in it (because every transaction has a dot somewhere in it), THEN it would show both transactions.  Then, if I backspace to clear the search bar, 1 of the transactions would immediately disappear again.  So, it was there, but not showing.

So, I added a label to the one that was showing, then used the search bar to make the other one show, and added a different label to that one.  Backspace the search bar, and 1 would disappear still.  However, now when I close and re-open suite, they both show up.

It's like maybe because they were the same address and same amount of coins, it was filtering 1 out until I gave them unique labels and then closed/opened suite again... but  I don't know if the 2 identical transactions actually have anything to do with it or not.

**Info:**
 - Remembered Wallet: YES
 - Suite Version: 21.5.1
 - Browser: Suite, not browser
 - OS: Windows 10 20H2
 - Bridge: 
 - Firmware Version: Model T, current firmware

**Steps to reproduce:**
Send x btc to address y
After 1 minute, send x btc to address y (use the SAME x and y both times)
See that both are pending
Wait for confirmation and see that 1 has disappeared

**Expected result:**
Both transactions should show up after confirmation.

**Actual result:**
One one transaction showed up, the other was "known" by suite because I could search it, but did not show up otherwise.

**Comment by uj** [comment]
Several more reports of this happening recently on forums.  

Transactions not showing up in desktop suite, but they show up in web site, for example.  

Here are just 3 examples I found with a quick search.  These are not my conversations, so I don't know any details for these, this is just to show that it is happening to multiple people and not just a one-time issue.

https://www.reddit.com/r/TREZOR/comments/obbhtp/wheres_my_bitcoin/

https://www.reddit.com/r/TREZOR/comments/oafltu/i_exchange_ltc_on_my_trezor_suite_and_its_not/

https://www.reddit.com/r/TREZOR/comments/o9qnz3/withdrew_some_eth_to_my_wallet_address_why_are/

**Comment by slush0** [comment]
I experienced this issue twice myself, during the last three weeks.

**Comment by jerechinsky** [comment]
@trezor/qa could you please try to reproduce?

**Comment by bosomt** [comment]
@alex-jerechinsky im not able to reproduce issue, yet, i still see same transaction history, 
i'm already working on it and i will update this comment with more results

already tested loaded Suite multiple times, including this week transactions.

```
21.8.1 web, backend: btc4.trezor.io, Trezor model One 
vs
21.9.1 app, backend: btc2.trezor.io, Trezor model T
```

**Comment by slush0** [comment]
It happen to me for various receiving addresses (no address reuse) and changes of the labeling (and restarting Suite) does not impact the behaviour in any way (tx is still missing). So I cannot confirm the observation of @uj that it is somehow relevant to the handling of transactions of the same address.

**Comment by slush0** [comment]
However, forgetting and replugging Trezor (triggering full account discovery) fixes this.

**Comment by bosomt** [comment]
**QA OK**

AFAIK issue will be fixed in 21.11.1 release that will be released today
https://github.com/trezor/trezor-suite/pull/4418
